### PR TITLE
Add Checkbox

### PR DESCRIPTION
**Pull request by kwood15**
Please look over this and refactor me ta

### CODE REVIEW

File: src/components/Form/CheckboxInput.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import T from 'prop-types'
+import { classNames } from '../../utils/'
```
Comment (rbrtsmith) [inline src/components/Form/CheckboxInput.jsx:3]: Good

File: src/components/Form/CheckboxInput.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+const CheckboxInput = ({ className, ...rest }) => (
+  <input className={classNames('c-form-input__input', className)} {...rest} />
```
Comment (rbrtsmith) [inline src/components/Form/CheckboxInput.jsx]: Good

File: src/components/Form/CheckboxInput.jsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import T from 'prop-types'
+import { classNames } from '../../utils/'
+
+const CheckboxInput = ({ className, ...rest }) => (
```
Comment (rbrtsmith) [inline src/components/Form/CheckboxInput.jsx]: Lets just call it `Checkbox`

File: src/components/Form/CheckboxInput.jsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  <input className={classNames('c-form-input__input', className)} {...rest} />
+)
```
Comment (rbrtsmith) [inline src/components/Form/CheckboxInput.jsx]: We're missing some bits of the checkbox.  In nebula it's hidden away and always comes with the associated label to style it etc.
screen shot 2017-08-08 at 13 27 46

HTML from Nebula Components:
```html
<div class="c-form-input">
  <input type="checkbox" class="c-form-input__input" id="check-1" value="on">
   <label for="check-1" class="c-form-input__label c-form-input__label--checkbox">
      Select
   </label>
</div>
```

**Review by rbrtsmith — CHANGES_REQUESTED** [review]

**Comment by kwood15** [comment]
Have refactored the checkboxes and added tests for the additional elements - 1 test currently broken so we can over this tmw if you like?

**Comment by kwood15** [comment]
Have refactored the tests and have added to demo so please review. Screenshot here:

screen shot 2017-08-11 at 13 22 43

File: src/components/Form/__tests__/Checkbox.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import React from 'react'
```
Comment (rbrtsmith) [inline src/components/Form/__tests__/Checkbox.jsx:1]: This test file is testing `CheckboxWrapper` and not `Checkbox`

File: src/components/Form/Label.jsx [hunk 6]
```diff
@@ -3,8 +3,8 @@ import T from 'prop-types'
 import classNames from '../../utils/classNames'
 
 const Label = ({ className, children, htmlFor, ...rest }) => (
-  <label htmlFor={htmlFor} className={classNames('c-label', className)} {...rest}>
-    { children }
+  <label id="c-label" htmlFor={htmlFor} className={classNames('c-label', className)} {...rest}>
```
Comment (rbrtsmith) [inline src/components/Form/Label.jsx]: Hard coded `id`.  Label's don't need an `id` so just let it be optionally passed with `...rest`

File: src/components/Form/__tests__/CheckboxWrapper.jsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+import { Form } from '../'
+
+describe('<Form.CheckboxWrapper id="checkbox-1" />', () => {
+  it('renders the appropriate classNames', () => {
+    const $ = shallow(<Form.CheckboxWrapper className="test" />)
+    expect($.hasClass('c-form-input test')).toBe(true)
+  })
+
+  it('renders children', () => {
```
Comment (rbrtsmith) [inline src/components/Form/__tests__/CheckboxWrapper.jsx:12]: Let's test that children get rendered in the correct order like in `RadioWrapper` as the ordering is critical for functionality.

File: src/components/Form/__tests__/Label.jsx [hunk 8]
```diff
@@ -15,8 +15,7 @@ describe('<Form.Label />', () => {
   })
 
   it('takes attributes', () => {
-    const $ = shallow(<Form.TextInput placeholder="test" id="foo" htmlFor="t" />)
-    expect($.prop('placeholder')).toBe('test')
+    const $ = shallow(<Form.Label id="foo" htmlFor="t" />)
```
Comment (rbrtsmith) [inline src/components/Form/__tests__/Label.jsx]: This test still passes despite `id` in `Label` being hard coded as the `...rest` spread in `Label` overrides it.  You can just remove the `id` prop from `Label`

**Review by rbrtsmith — CHANGES_REQUESTED** [review]
Really good work so far.  Just a few things need fixing then this is good to merge in.

File: src/components/Form/__tests__/Checkbox.jsx (hunk 5 above)
Comment (kwood15) [reply src/components/Form/__tests__/Checkbox.jsx:1]: Done

File: src/components/Form/Label.jsx (hunk 6 above)
Comment (kwood15) [reply src/components/Form/Label.jsx]: Done

File: src/components/Form/__tests__/CheckboxWrapper.jsx (hunk 7 above)
Comment (kwood15) [reply src/components/Form/__tests__/CheckboxWrapper.jsx:12]: Done on the Checkbox test

File: src/components/Form/__tests__/Label.jsx (hunk 8 above)
Comment (kwood15) [reply src/components/Form/__tests__/Label.jsx]: Done

**Comment by kwood15** [comment]
Have fixed the above

**Review by rbrtsmith — APPROVED** [review]

**Pull request merged**